website/photos/management/commands/link_album_events.py

```diff
@@ -0,0 +1,111 @@
+from django.core.management.base import BaseCommand
+
+from events.models import Event
+from photos.models import Album
+
+import csv
+
+
+def write(events, albums, file):
+    for j in range(max(len(events), len(albums))):
+        if j < len(events):
+            file.write(
+                str(events[j].pk)
+                + "-"
+                + events[j].title
+                + "-"
+                + str(events[j].start.date())
+            )
+        file.write(",")
+        if j < len(albums):
+            file.write(
+                str(albums[j].pk) + "-" + albums[j].title + "-" + str(albums[j].date)
+            )
+        file.write("\n")
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        events = Event.objects.order_by("start").all()
+        albums = Album.objects.order_by("date").all()
+
+        # partition the events and albums by date
+        partitioned_events = [[events[0]]]
+        last_date = events[0].start.date()
+        for e in events[1:]:
+            if e.start.date() == last_date:
+                partitioned_events[len(partitioned_events) - 1].append(e)
+            else:
+                partitioned_events.append([e])
+                last_date = e.start.date()
+
+        partitioned_albums = [[albums[0]]]
+        last_date = albums[0].date
+        for a in albums[1:]:
+            if a.date == last_date:
+                partitioned_albums[len(partitioned_albums) - 1].append(a)
+            else:
+                partitioned_albums.append([a])
+                last_date = a.date
+
+        print("Writing to link_album_events.csv")
+        i = 0  # index of partitioned_events
+        j = 0  # index of partitioned_albums
+        f = open("link_album_events.csv", "w")
```

Comment:
I think it is better to just print the output to stdout, and read from stdin. Then the user can do the IO via the shell application just as they like, which is more versatile.

Reply:
For output that's not really nice right, because you can't print updates or instructions. And then probably nice to do the same for input. 
Maybe just adding an option to supply the file location would be better?

Reply:
@pingiun what do you think? You know best how to do this on production

Reply:
For our current production environment this would probably work fine, but a file does make it harder to execute. You have to download a file, then change it, then upload the file again. And your server is remote and you usually only have a remote shell unless you start working with scp.

Reply:
I think the best way is to print the csv data output on stdout and print informational messages on stderr. Then the importing step can take input from stdin.

Reply:
I didn't see this last comment, I put the information in help